cmd/kafka-consumer/writer.go

```diff
@@ -379,31 +386,24 @@ func (w *writer) WriteMessage(ctx context.Context, message *kafka.Message) bool
 			w.checkPartition(row, partition, message.TopicPartition.Offset)
 
 			tableID := row.GetTableID()
-			if w.option.protocol != config.ProtocolSimple {
+			switch w.option.protocol {
+			case config.ProtocolSimple, config.ProtocolCanalJSON:
```

Comment:
Why exclude open protocol?

Reply:
Simple and Canal-JSON, the table id is already encoded into the message.

Open-Protocol, table id is only set for the partition table, so we have to generate fake table id for all open protocol encoded messages.